packages/account/src/BiconomySmartAccountV2.ts

```diff
@@ -499,11 +507,75 @@ export class BiconomySmartAccountV2 extends BaseSmartContractAccount {
     return encodeAbiParameters(parseAbiParameters("bytes, address"), [moduleSignature, moduleAddressToUse]);
   }
 
+  /**
+   * Sets paymaster-related fields in the provided user operation based on the specified paymaster service data.
+   *
+   * @param {Partial<UserOperationStruct>} userOp - The partial user operation structure to be modified.
+   * @param {PaymasterUserOperationDto} paymasterServiceData - The paymaster service data containing mode and additional information.
+   * @returns {Promise<Partial<UserOperationStruct>>} A promise that resolves to the modified user operation structure.
+   */
+  async setPaymasterFields(
+    userOp: Partial<UserOperationStruct>,
+    paymasterServiceData: PaymasterUserOperationDto,
+  ): Promise<Partial<UserOperationStruct>> {
+    if (this.paymaster !== undefined) {
+      try {
+        if (paymasterServiceData.mode === PaymasterMode.SPONSORED) {
+          const paymasterData = await (this.paymaster as IHybridPaymaster<PaymasterUserOperationDto>).getPaymasterAndData(userOp, {
+            mode: paymasterServiceData.mode,
+          });
+          userOp.paymasterAndData = paymasterData.paymasterAndData;
+          userOp.callGasLimit = paymasterData.callGasLimit;
+          userOp.verificationGasLimit = paymasterData.verificationGasLimit;
+          userOp.preVerificationGas = paymasterData.preVerificationGas;
+          return userOp;
+        } else if (paymasterServiceData.mode === PaymasterMode.ERC20) {
```

Comment:
if the mode is ERC20 and if a preferred token in given in paymasterServiceData then we keep this part here only for this specific case. Downside still is that user won't be able to see expected fee quote if it's part of ux flow.

Reply:
This was fixed